lib/internal/test_runner/tap_stream.js

```diff
@@ -48,26 +50,22 @@ class TapStream extends Readable {
     }
   }
 
-  bail(message) {
-    this.#tryPush(`Bail out!${message ? ` ${tapEscape(message)}` : ''}\n`);
+  fail(nesting, testNumber, name, details, directive) {
+    this.emit('test:fail', { __proto__: null, name, nesting, testNumber, details, ...directive });
```

Comment:
These events should probably not be emitted from here anymore and only emitted from `reporter_stream`. There is no harm in emitting them here, but the test runner shouldn't rely on anything coming from reporters that could potentially be from userland.

Reply:
I really think we can remove the events, there were really only needed before there were reporters